tests/publiccloud/check_registercloudguest.pm

```diff
@@ -39,80 +39,80 @@ sub run {
         # all BYOS related checks. So we just regestering system and going further
         registercloudguest($instance);
     } elsif (is_byos()) {
-        if ($instance->run_ssh_command(cmd => 'sudo zypper lr', proceed_on_failure => 1) !~ /No repositories defined/gm) {
+        if ($instance->ssh_script_output(cmd => 'sudo zypper lr', proceed_on_failure => 1) !~ /No repositories defined/gm) {
```

Comment:
Do we really want `proceed_on_failure => 1` here?

Reply:
same idea for all similar questions below . won't spam PR with duplicating it

Reply:
Which `zypper` errors can occur when listing the local repositories?

Reply:
```
$ zypper lr; echo $?
Warning: No repositories defined.
Use the 'zypper addrepo' command to add one or more repositories.
6
```

Reply:
I have no idea , zypper may simply crash on calling list repo command just because poorly written code or some library missing , there could be billions reasons why package manager will throw error . And the point that here we want to fail only in one very certain situation . Because we not testing zypper in this certain test but we using zypper as testing tool allowing to prove that there is no repos nothing more than that

Reply:
> ```shell
> $ zypper lr; echo $?
> Warning: No repositories defined.
> Use the 'zypper addrepo' command to add one or more repositories.
> 6
> ```

Does it make sense to continue in such a case? I though the check for empty repositories is exactly what we are doing here, no?

> I have no idea , zypper may simply crash on calling list repo command just because poorly written code or some library missing , there could be billions reasons why package manager will throw error . And the point that here we want to fail only in one very certain situation . Because we not testing zypper in this certain test but we using zypper as testing tool allowing to prove that there is no repos nothing more than that

Yes but if `zypper` is broken then there is no point in continuing because "ex falso quodlibet".

Keep in mind: This is a local zypper call without network interaction, so the chance of sporadic failures due to network issues, outdated repositories, SCC, ... does not happen here. That's why I don't see why we should make a bad compromise **at this place**. If it would be a `zypper ref` or something else, we wouldn't have this argument :slightly_smiling_face:

Reply:
@asmorodskyi @grisu48 please let me know what I should do here.

Reply:
From my point of view: Simply remove the `proceed_on_failure => 1`, if possible.